noxfile.py

```diff
@@ -63,19 +62,7 @@ def run_coverage(session):
     set_environment_variables(PYBAMM_ENV, session=session)
     session.install("coverage", silent=False)
     if sys.platform != "win32":
```

Comment:
We can now remove the `sys.platform != "win32"` checks from everywhere in the file because all dependencies will now support all platforms

The only condition we will still need should be the Python <3.9 version check (where JAX is not available)

Reply:
Since that change is related to enabling JAX on windows and not removing ODES, I think that should be a different PR. I can still do it though

Reply:
JAX is already enabled on Windows right now, so feel free to remove it here